programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -428,14 +429,19 @@ fn translate_string_and_do(
     len: u64,
     check_aligned: bool,
     check_size: bool,
+    stop_truncating_strings_in_syscalls: bool,
     work: &mut dyn FnMut(&str) -> Result<u64, EbpfError>,
 ) -> Result<u64, EbpfError> {
     let buf = translate_slice::<u8>(memory_mapping, addr, len, check_aligned, check_size)?;
-    let i = match buf.iter().position(|byte| *byte == 0) {
-        Some(i) => i,
-        None => len as usize,
+    let msg = if stop_truncating_strings_in_syscalls {
+        buf
+    } else {
+        let i = match buf.iter().position(|byte| *byte == 0) {
+            Some(i) => i,
+            None => len as usize,
+        };
+        buf.get(..i).ok_or(SyscallError::InvalidLength)?
```

Comment:
length check is still handled in `translate_slice()` after activation, right?

Reply:
You mean the bounds check to prevent it from leaving its guest memory mapping?
Yes, that is handled here: https://github.com/solana-labs/solana/blob/a9004fdfc1a2ec54619fa99d6b3f03fe39e096d3/programs/bpf_loader/src/syscalls/mod.rs#L334

Reply:
yep. thanks!